### PR TITLE
feat: implements badge on tab component

### PR DESCRIPTION
**Pull request by Atzac**

### CODE REVIEW

File: src/components/tab/tab.test.tsx [hunk 1]
```diff
@@ -11,11 +11,14 @@ const defaultTab: TabProps = {
...
 
-function sut(props: TabProps = defaultTab): HTMLElement {
+function sut(props: TabProps = defaultTab) {
   render(<IonTab {...props} />);
-  return screen.getByRole('button', { name: props.label });
 }
 
+const getTab = () => {
+  return screen.getByTestId('ion-tab');
+};
```
Comment (jairochaves2) [inline src/components/tab/tab.test.tsx:20]:
```suggestion
const getTab = (id='ion-tab') => {
  return screen.getByTestId(id);
};
```

To reuse in lines 125 and 131

**Review by jairochaves2 — APPROVED** [review]

**Review by wermesonsl — APPROVED** [review]

File: src/components/tab/tab.test.tsx [hunk 2]
```diff
@@ -25,28 +28,31 @@ describe('Tab', () => {
...
 
     it('should be disabled', () => {
-      expect(sut({ ...defaultTab, disabled: true })).toHaveAttribute(
-        'disabled'
-      );
+      sut({ ...defaultTab, disabled: true });
+      expect(getTab()).toHaveAttribute('disabled');
     });
 
     it('should render tab without disabled by default', async () => {
```
Comment (iurynogueira) [inline src/components/tab/tab.test.tsx:48]:
```suggestion
    it('should render tab enabled by default', async () => {
```

File: src/components/tab/tab.test.tsx [hunk 3]
```diff
@@ -99,4 +107,28 @@ describe('Tab', () => {
...
+
+  describe('With Badge', () => {
+    it('should render badge', async () => {
+      sut({ ...defaultTab, badge: { label: 2 } });
+      expect(screen.getAllByTestId('ion-badge'));
+    });
+
+    it('should not render badge when is not informed', async () => {
+      sut();
+      expect(screen.queryAllByText('ion-badge')).toHaveLength(0);
```
Comment (iurynogueira) [inline src/components/tab/tab.test.tsx:119]: screen.queryAllByText('ion-badge') this is correct? I believe that you did should find by badge label or by testid.

File: src/components/tab/tab.test.tsx [hunk 4]
```diff
@@ -99,4 +107,28 @@ describe('Tab', () => {
...
+    });
+
+    it('should render badge with string label', async () => {
+      const label = 'Skywalker';
+      sut({ ...defaultTab, badge: { label: label } });
+      expect(screen.getByTestId('ion-badge').textContent).toBe(label);
```
Comment (iurynogueira) [inline src/components/tab/tab.test.tsx:125]:
```suggestion
      expect(screen.getByTestId('ion-badge').textContent).toBe(label);
```

You can create a function like getTab.

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**